pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/TypeHelper.java

```diff
@@ -45,21 +65,6 @@ public static boolean subclasses(TypeNode n, Class<?> clazz) {
             return clazz.getSimpleName().equals(((Node) n).getImage()) || clazz.getName().equals(((Node) n).getImage());
         }
 
-        if (type.equals(clazz)) {
-            return true;
-        }
-
-        List<Class<?>> implementors = Arrays.asList(type.getInterfaces());
-        if (implementors.contains(clazz)) {
-            return true;
-        }
-        Class<?> superC = type.getSuperclass();
-        while (superC != null && !superC.equals(Object.class)) {
-            if (superC.equals(clazz)) {
-                return true;
-            }
-            superC = superC.getSuperclass();
-        }
-        return false;
+        return clazz.isAssignableFrom(type);
```

Comment:
Not sure why it wasn't this way before, considering this is a native (and much faster) method; and the old code was broken when dealing with an interface that extends another one (`getInterfaces()` returns only **direct** implemented interfaces); and when the interface was implemented by a super class.